apps/remote_control/lib/lexical/remote_control/analyzer.ex

```diff
@@ -1,15 +1,43 @@
 defmodule Lexical.RemoteControl.Analyzer do
   alias Lexical.Ast
   alias Lexical.Ast.Analysis
+  alias Lexical.Ast.Analysis.Require
+  alias Lexical.Ast.Analysis.Use
   alias Lexical.Document.Position
   alias Lexical.RemoteControl.Analyzer.Aliases
   alias Lexical.RemoteControl.Analyzer.Imports
+  alias Lexical.RemoteControl.Analyzer.Requires
+  alias Lexical.RemoteControl.Analyzer.Uses
 
   require Logger
 
   defdelegate aliases_at(analysis, position), to: Aliases, as: :at
   defdelegate imports_at(analysis, position), to: Imports, as: :at
 
+  @spec requires_at(Analysis.t(), Position.t()) :: [module()]
+  def requires_at(%Analysis{} = analysis, %Position{} = position) do
+    analysis
+    |> Requires.at(position)
+    |> Enum.reduce([], fn %Require{} = require, acc ->
+      case expand_alias(require.module, analysis, position) do
+        {:ok, expanded} -> [expanded | acc]
+        _ -> [Module.concat(require.as) | acc]
+      end
+    end)
+  end
+
+  @spec uses_at(Analysis.t(), Position.t()) :: [module()]
+  def uses_at(%Analysis{} = analysis, %Position{} = position) do
+    analysis
+    |> Uses.at(position)
+    |> Enum.reduce([], fn %Use{} = use, acc ->
+      case expand_alias(use.module, analysis, position) do
+        {:ok, expanded} -> [expanded | acc]
+        _ -> [Module.concat(use.module) | acc]
+      end
+    end)
+  end
```

Comment:
It looks like the `.module` for aliases/imports are expanded during analysis, which means we can just use a simple `defdelegate` for those, but not for requires/uses? Is there a reason we can't expand the module before the `%Require{}/%Use{}` is created in `analysis.ex`? It seems like we should either do all expansion there or none of the expansion there (i.e. refactor analyzing aliases/imports so that it's consistent).

Reply:
i was slightly worried about performance, but I think it's important to be consistent.

Reply:
Actually, depending on what you mean by expanded, I don't think this is correct. The aliases are still a list of atoms, and not a module at this point. Calling expand_alias on them makes them into a module. Same thing with aliases, but they have a `to_module` function in them that I didn't think was necessary, since aliases and imports return maps when requires and uses just return a simple list.